### PR TITLE
Adding ascidoc documentation

### PR DESCRIPTION
**Pull request by manolo**
Connected to #1

[Review on Reviewable](https://reviewable.io/reviews/vaadin/vaadin-upload/26)

### CODE REVIEW

File: docs/vaadin-upload-overview.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+The [elementname]#vaadin-date-picker# is a Polymer element for uploading multiple files to server.
+
+It allows users to add their files by dropping them directly into the component. Also it has the “Add files” button, which opens the native browser file dialog for selecting the file or files to process.
+
+All files added by user are displayed in a file list inside the element.
+
+The added files would be automatically uploaded to server. The uploading of multiple files is performed simultaneously. The list indicates the state and uploading progress for each file.
+
+Each file could be removed from the list by user by pressing the corresponding “×” icon. In this case, if the file uploading has been in progress, then the upload process would be aborted.
```
Comment (jounik) [inline docs/vaadin-upload-overview.adoc]: I would not say "x" icon (avoid describing visual styles in the documentation). Instead, try "by pressing the corresponding cancel button".

**Comment by manolo** [comment]
Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[docs/vaadin-upload-overview.adoc, line 18 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KAAcfD5LCuNVROs-DQH-r1-18:-KAAphHCtyNCIBo0Uj7_:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/6068bd97dee1185d64cb57a3a989a1224455421c/docs/vaadin-upload-overview.adoc#L18)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26)*

**Comment by manolo** [comment]
@magi42 did some re-wording and formatting based in your comments from #6

**Comment by magi42** [comment]
Reviewed 1 of 4 files at r2.
Review status: 1 of 5 files reviewed at latest revision, 15 unresolved discussions.

---

*[docs/vaadin-upload-basic.adoc, line 11 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABdWnnYI5HjxeGav_K:-KABdWnnYI5HjxeGav_L:-2105189686) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L11)):*
"by" missing between "files [by] opening" and "or [by] dropping".
Saying "files" also covers one file here without much problems.
I'd rather rephrase it a little bit:
"The user can select files to upload by using the file selector in the browser or by dropping the files from the desktop onto the component."
A bit long sentence though. I'm not sure if it should be "onto" or "into" (which is used in another file).

---

*[docs/vaadin-upload-basic.adoc, line 13 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABhQiSDoXI1_v2TmxW:-KABhQiSDoXI1_v2TmxX:1695176547) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L13)):*
I'd rather use "by using" here, although also just "using" is probably OK. The difference is rather subtle and I'm not quite sure which fits here.

---

*[docs/vaadin-upload-basic.adoc, line 22 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABiRSRf5rRz83aKFW6:-KABiRSSNCwDFV1t40YN:1909702447) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L22)):*
* Use `[vaadinelement]#vaadin-upload#` to emphasize Vaadin elements (products). Use `[elementname]#...#` for other HTML elements.
* You need to specify the category of vaadin-upload: "The vaadin-upload element..." ("vaadin-upload" is a name rather than category and names can not have articles).
* Use "or" in the list: "...added, removed, or rejected, ...". The as well as is an additional thing after the list. It's a bit odd though, as "as well as" is sort of "and", not "or".
* per -> _for each_

---

*[docs/vaadin-upload-basic.adoc, line 23 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABjhC1MC9cQ3zF31iz:-KABjhC1MC9cQ3zF31j-:-1289320262) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L23)):*
* those -> the
* ...or _for_ preventing...

---

*[docs/vaadin-upload-basic.adoc, line 24 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABjs4cvsJF_1SYGJQR:-KABjs4cvsJF_1SYGJQS:1664399124) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L24)):*
* Maybe "complete" sounds better than "full"
* What's a "description" here? Is is something comparable to an event? Or did you mean descriptions of the events? Perhaps say "..._and their descriptions_..."
* read -> _refer to_

---

*[docs/vaadin-upload-basic.adoc, line 27 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABkoNTWoRkbofePXns:-KABkoNTWoRkbofePXnt:-758183374) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L27)):*
* contains -> _contain_
* Use the`[propertyname]#event.detail.file#` style name

---

*[docs/vaadin-upload-basic.adoc, line 28 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABl12W93AZMmiH6UTY:-KABl12XM8MwGQrinHcx:315661596) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L28)):*
* Use comma after the introductory clause: "Additional, ..."
* Use the `[propertyname]#event.detail.xhr#` style
* Articles: I think the property name requires "a" and the class name either a or the, not sure about the meaning.

---

*[docs/vaadin-upload-basic.adoc, line 30 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABmKoEm833A2DwJvJo:-KABmKoFIyfHMjc6pSUs:-2033285665) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L30)):*
* Use "the following" rather than "this", which makes hard to say where "this" is (and is also maybe informal)
* Comma after introductory clause: "In this example, we..."
* "...events, _which are_ fired..."

---

*[docs/vaadin-upload-basic.adoc, line 40 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABn-HkoreuTJ1KIX8i:-KABn-HlpPigic_8ikIv:772769105) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L40)):*
* I'd drop the "also" from the beginning and perhaps (but not necessarily) have it later: "...you can also use..."
* Why is `Polymer` in courier here? It's not an identifier but a regular name.

---

*[docs/vaadin-upload-overview.adoc, line 18 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KAAcfD5LCuNVROs-DQH-r1-18:-KABpYBygbo8GYuVUQ55:-492831204) ([raw file](https://github.com/vaadin/vaadin-upload/blob/798745ddc02c0a8151473c59e2aec864e5983f27/docs/vaadin-upload-overview.adoc#L18)):*
Hmm... Emphasizing the `cancel/remove` with courier looks a bit odd as well, as that would indicate a technical (code) meaning of the words, which they are not, and the are not captions. Rather, maybe just talk about "cancel button" as a concept and trust that the reader understands which it is.
Also, the slash is a bit odd, as is the "each file", as the user doesn't typically remove all the files. Maybe rather use active voice.
Maybe: "The user can cancel the upload of a file and remove it from the list by pressing the cancel button for the particular file."

---

*[docs/vaadin-upload-overview.adoc, line 10 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABofkM3RBmDmlxVVjV:-KABofkNN1s1k56Xk_yz:1440680478) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-overview.adoc#L10)):*
I think we're talking about `vaadin-upload` in this file, not `vaadin-date-picker`. ;-)

---

*[docs/vaadin-upload-overview.adoc, line 13 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABpLi1y2vfMnP4ql72:-KABpLi1y2vfMnP4ql73:-719847508) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-overview.adoc#L13)):*
* Maybe "a file" rather than "the file"

---

*[docs/vaadin-upload-overview.adoc, line 22 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABqDgjCWqKaVF1Wc5B:-KABqDgk4dNkepH7EFA9:1005876668) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-overview.adoc#L22)):*
* Maybe add: "If _the button_ is pressed..."
* "would be" -> "will be". In [conditional clauses](http://www.edufind.com/english-grammar/type-1-conditional/), the condition is in present tense, while the result is in normal future tense: if is then will be.

---

*[docs/vaadin-upload-overview.adoc, line 26 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KAC9nYs2ADHplOm9OTg:-KAC9nYs2ADHplOm9OTh:-2056585360) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-overview.adoc#L26)):*
* I think you can use `[vaadinelement]#vaadin-upload#` here as well
* add the...element: "of _the_ ... _element_"

---

*[docs/vaadin-upload-overview.adoc, line 27 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KACDFic54q_CD7o34rh:-KACDFidK4Ujt4FQzcwD:1809059887) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-overview.adoc#L27)):*
There seems to be a some sort of retry button there, but it's not mentioned in the text.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26)*

**Comment by manolo** [comment]
Review status: 1 of 5 files reviewed at latest revision, 15 unresolved discussions.

---

*[docs/vaadin-upload-basic.adoc, line 11 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABdWnnYI5HjxeGav_K:-KADvhxDxfX6icwvA4Dm:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L11)):*
Done.

---

*[docs/vaadin-upload-basic.adoc, line 13 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABhQiSDoXI1_v2TmxW:-KADvq8jSDePJV2e2Mki:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L13)):*
Done.

---

*[docs/vaadin-upload-basic.adoc, line 22 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABiRSRf5rRz83aKFW6:-KADwSTk5hZvlIKrO-u5:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L22)):*
Done.

---

*[docs/vaadin-upload-basic.adoc, line 23 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABjhC1MC9cQ3zF31iz:-KADwYVLmOXVypkLPTzi:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L23)):*
Done.

---

*[docs/vaadin-upload-basic.adoc, line 24 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABjs4cvsJF_1SYGJQR:-KADwkXd7kM2EXwZhWWL:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L24)):*
Done.

---

*[docs/vaadin-upload-basic.adoc, line 27 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABkoNTWoRkbofePXns:-KADwuHPj2mNxLMmEx9o:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L27)):*
Done.

---

*[docs/vaadin-upload-basic.adoc, line 28 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABl12W93AZMmiH6UTY:-KADxHqn_bA_RAAN0cLp:1452520942) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L28)):*
I think `with the .corresponding` is correct. Done.

---

*[docs/vaadin-upload-basic.adoc, line 30 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABmKoEm833A2DwJvJo:-KADxrarVqwXm6ybVNZ9:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L30)):*
Done.

---

*[docs/vaadin-upload-basic.adoc, line 40 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABn-HkoreuTJ1KIX8i:-KADy4iRdKD57vmCjIJq:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L40)):*
Done.

---

*[docs/vaadin-upload-overview.adoc, line 18 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KAAcfD5LCuNVROs-DQH-r1-18:-KADyrwjzJK5HmiyxgdX:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/798745ddc02c0a8151473c59e2aec864e5983f27/docs/vaadin-upload-overview.adoc#L18)):*
Done.

---

*[docs/vaadin-upload-overview.adoc, line 10 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABofkM3RBmDmlxVVjV:-KADyAUSu08HdHAuHqQj:1839419229) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-overview.adoc#L10)):*
good catch. Done.

---

*[docs/vaadin-upload-overview.adoc, line 13 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABpLi1y2vfMnP4ql72:-KADyWmsHFbOgcNNrYBo:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-overview.adoc#L13)):*
Done.

---

*[docs/vaadin-upload-overview.adoc, line 22 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABqDgjCWqKaVF1Wc5B:-KADz7H6L5ELn7TfzEf4:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-overview.adoc#L22)):*
Done.

---

*[docs/vaadin-upload-overview.adoc, line 26 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KAC9nYs2ADHplOm9OTg:-KADzMD1UJ0dVIm2lUgx:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-overview.adoc#L26)):*
Done.

---

*[docs/vaadin-upload-overview.adoc, line 27 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KACDFic54q_CD7o34rh:-KAE-GGbwqKTVGP2e0HW:1328982371) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-overview.adoc#L27)):*
Added a sentence for that. Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26)*

**Comment by magi42** [comment]
Review status: 1 of 5 files reviewed at latest revision, 5 unresolved discussions.

---

*[docs/vaadin-upload-basic.adoc, line 22 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABiRSRf5rRz83aKFW6:-KAE9Tk8q9OyjP6wQkj_:-997020093) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L22)):*
I've normally used [serial comma](https://en.wikipedia.org/wiki/Serial_comma) in lists: "added, removed, or rejected". It is recommended in most English style guides, especially for American English.

---

*[docs/vaadin-upload-basic.adoc, line 23 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABjhC1MC9cQ3zF31iz:-KAEAaTbfV5JfI1wkTjz:-669506200) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L23)):*
Ah, it's an either-a-or-b structure, which [should not have a comma](http://www.grammarly.com/handbook/punctuation/comma/29/comma-between-correlative-conjunction-sets/) before the "or".

---

*[docs/vaadin-upload-basic.adoc, line 28 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABl12W93AZMmiH6UTY:-KAEGN0VPZ1fNIj8ghjN:-3623357) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L28)):*
Ah, "event" starts with a vocal, so it should be "an"... Those style names cause tricks to the brain. ;-)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26)*

**Comment by manolo** [comment]
Review status: 1 of 5 files reviewed at latest revision, 4 unresolved discussions.

---

*[docs/vaadin-upload-basic.adoc, line 22 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABiRSRf5rRz83aKFW6:-KAELqpkp-zzc16ABBtD:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L22)):*
Done.

---

*[docs/vaadin-upload-basic.adoc, line 23 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABjhC1MC9cQ3zF31iz:-KAELxk85AVvTiPqeOnY:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L23)):*
Done.

---

*[docs/vaadin-upload-basic.adoc, line 28 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26#-KABl12W93AZMmiH6UTY:-KAEMHl24yQwNNLOgbFb:-499146009) ([raw file](https://github.com/vaadin/vaadin-upload/blob/1ab21a9c5762863795202ab3a73d940bc20bebc8/docs/vaadin-upload-basic.adoc#L28)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26)*

**Comment by magi42** [comment]
Reviewed 1 of 4 files at r1, 1 of 4 files at r2, 2 of 2 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-upload/26)*

**Pull request merged**